apps/api/src/app/events/usecases/parse-event-request/parse-event-request.usecase.ts

```diff
@@ -206,4 +221,68 @@ export class ParseEventRequest {
 
     return reservedVariables?.map((reservedVariable) => reservedVariable.type) || [];
   }
+
+  public async sendInAppNudgeForTeamMemberInvite(command: ParseEventRequestCommand) {
+    // check if this is first trigger
+    const notifications = await this.notificationRepository.count({
+      _organizationId: command.organizationId,
+      _environmentId: command.environmentId,
+    });
+
+    if (notifications > 0) return;
+
+    // check if user is using personal email
+    const user = await this.userRepository.findOne({
+      _id: command.userId,
+    });
+
+    if (this.checkEmail(user?.email)) return;
+
+    // check if organization has more than 1 member
+    const membersCount = await this.memberRepository.count({
+      _organizationId: command.organizationId,
+    });
+
+    if (membersCount > 1) return;
+
+    if ((process.env.NODE_ENV === 'dev' || process.env.NODE_ENV === 'production') && process.env.NOVU_API_KEY) {
+      if (!command.payload[INVITE_TEAM_MEMBER_NUDGE_PAYLOAD_KEY]) {
+        const novu = new Novu(process.env.NOVU_API_KEY);
+
+        novu.trigger(
+          process.env.NOVU_INVITE_TEAM_MEMBER_NUDGE_TRIGGER_IDENTIFIER || 'in-app-invite-team-member-nudge',
+          {
+            to: {
+              subscriberId: command.userId,
+              email: user?.email as string,
+            },
+            payload: {
+              INVITE_TEAM_MEMBER_NUDGE_PAYLOAD_KEY: true,
+              webhookUrl: `${process.env.API_ROOT_URL}/v1/invites/webhook`,
+            },
```

Comment:
❓ **question:**shouldn't this be like:
```suggestion
            payload: {
              [INVITE_TEAM_MEMBER_NUDGE_PAYLOAD_KEY]: true,
              ...
            },
```

Reply:
I feel it should work without [], because `INVITE_TEAM_MEMBER_NUDGE_PAYLOAD_KEY` is of type string.
I will test it again. 🙂

Reply:
it won't, it will create an object like this: `{ INVITE_TEAM_MEMBER_NUDGE_PAYLOAD_KEY: true }` but we want it to be like `{ nv-type-team-member-invite-nudge: true }`